macros/extract_descriptor.sql

```diff
@@ -1,4 +1,43 @@
 -- grab descriptor codes from namespaced descriptor values
 {% macro extract_descriptor(col) -%}
-    split_part({{ col }}, '#', -1)
+    
+    {%- set stripped_col = col.split(":")[1] -%}
```

Comment:
I found one case where this doesn't work: `v:period:assessmentPeriodDescriptor::string`

We could maybe live with this, but I'm curious whether we can count from the end with negative subsets? These should always end with `descriptor_name::string` so maybe `[-3]` would do it?

Reply:
good catch. I think the `[-3]` would break when we call `extract_descriptor` from `extract_extension` here https://github.com/edanalytics/edu_edfi_source/blob/5c27a6b0e16c5d28e8108ebd0708590f17e9db4a/macros/extract_extension.sql#L30

but we could edit that ^ code to always cast to string? Or maybe add code to `extract_descriptor()` that adds on `::string` if it isn't there?

Reply:
latest commit addresses this. Also, I decided to make `replace_with` configurable to whatever column from `stg_ef3__descriptors`, so you can choose short vs. long description.

Tested with this config:
```
  descriptors:
    raceDescriptor:
      replace_with: short_description
    assessmentPeriodDescriptor:
      replace_with: description
    adaEligibilityDescriptor:
      replace_with: short_description
      
      
  extensions:
    stg_ef3__student_school_associations:
      ada_eligibility_descriptor:
        name: "tx:adaEligibilityDescriptor"
        dtype: 'string'
        extract_descriptor: True      
```